lib/facility.rb

```diff
@@ -1,14 +1,61 @@
+
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address,
+              :phone,
+              :services,
+              :registered_vehicles,
+              :collected_fees
+
+  # facility_details = { :name, :address, :phone}
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_details)
+    @name = facility_details [:name]
+    @address = facility_details [:address]
+    @phone = facility_details [:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?('Vehicle Registration')
+      vehicle.register
+      @registered_vehicles << vehicle
+        if vehicle.engine == :ev
+          @collected_fees += 200
+        elsif vehicle.plate_type == :antique
+          @collected_fees += 25
+        else
+          @collected_fees += 100
+        end
+      else
+    end
+  end
```

Comment:
Nice! I really like that you created a `register` method in your `Vehicle` class to manage setting the registration date and also assigning a plate type. I think that you're demonstrating the OOP pillar of Abstraction by doing this -- keeping logic in classes that only that class needs to know about.

Reply:
In your survey, you mentioned wanting some examples of how to DRY up your code. On lines 31, 33, and 35, you have very similar logic happening. You could, create a method, `collect_fee(amount)`, and have that logic in there. And then on the lines I listed, you could call to the method to perform the logic.